src/libmongoc/tests/TestSuite.c

```diff
@@ -506,8 +507,9 @@ TestSuite_RunFuncInChild (TestSuite *suite, /* IN */
    if (suite->mock_server_log_buf) {
       close (pipefd[1]);
       while ((nread = read (pipefd[0], buf, sizeof (buf) - 1)) > 0) {
-         buf[nread] = '\0';
-         mcommon_string_append (suite->mock_server_log_buf, buf);
+         mcommon_string_append_t append;
+         mcommon_string_append_init (&append, suite->mock_server_log_buf);
+         mcommon_string_append_bytes (&append, buf, nread);
```

Comment:
```suggestion
         mcommon_string_append_bytes (suite->mock_server_log_buf, buf, nread);
```
See suggestion above regarding `TestSuite::mock_server_log_buf`.